integration_tests/e2e/achieve-goal.cy.ts

```diff
@@ -0,0 +1,75 @@
+import DataGenerator from '../support/DataGenerator'
+import { PlanType } from '../../server/@types/PlanType'
+import PlanSummary from '../pages/plan-summary'
+
+describe('Achieve goal', () => {
+  const planSummary = new PlanSummary()
+
+  beforeEach(() => {
+    cy.createSentencePlan().then(planDetails => {
+      cy.wrap(planDetails).as('plan')
+      cy.openSentencePlan(planDetails.oasysAssessmentPk)
+    })
+  })
+
+  describe('Rendering', () => {
+    beforeEach(() => {
+      cy.get<{ plan: PlanType }>('@plan').then(({ plan }) => {
+        cy.addGoalToPlan(plan.uuid, DataGenerator.generateGoal()).then(goal => {
+          cy.wrap(goal).as('newGoal')
+          cy.addStepToGoal(goal.uuid, DataGenerator.generateStep())
+        })
+
+        planSummary.agreePlan()
+        cy.get('a').contains('Mark as achieved').click()
+      })
+    })
+
+    /* eslint-disable func-names */
+    // using function is the only way to access the wrapped newGoal - not available via arrow functions
+    it('Display agree plan page correctly on load', function () {
+      cy.get('h1').contains('has achieved this goal')
+      cy.get('.govuk-summary-card__title').should('contain', this.newGoal.title)
+    })
```

Comment:
This is one of the reasons I really dislike cypress. Anyway!
You can also do

      cy.get<{ goal: GoalType }>('@newGoal').then(({ goal }) => {
           // ... do stuff with goal in here
        })

Reply:
AS discussed on Slack I'll make this test an arrow function again and swap out this line for:

```cy.get<Goal>('@newGoal').then(goal => {
        cy.get('.govuk-summary-card__title').should('contain', goal.title)
      })```